cosmo_tester/test_suites/cli/linux_test.py

```diff
@@ -44,11 +50,180 @@ def test_cli_install_flow_linux(linux_cli_tester, logger):
+                       logger)
+
+
+def test_cfy_logs_linux_cluster(linux_cluster_cli_tester, logger):
+    cli_host = linux_cluster_cli_tester['cli_host']
+    nodes = linux_cluster_cli_tester['nodes']
+    paths = linux_cluster_cli_tester['paths']
+
+    # stop manager services so the logs won't change during the test
+    for node in nodes:
+        node.run_command('cfy_manager stop')
+
+    logs_dump_filepaths = json.loads(cli_host.run_command(
+        '{cfy} logs download --all-nodes --json'.format(cfy=paths['cfy'])
+    ).stdout.strip())['archive paths']
+
+    # assert all logs are downloaded
+    assert len(logs_dump_filepaths['manager']) == 3
+    assert len(logs_dump_filepaths['db']) == 3
+    assert len(logs_dump_filepaths['broker']) == 3
+
+    for node in nodes:
+        log_hashes = [f.split('  ')[0] for f in node.run_command(
+            'find /var/log/cloudify -type f -exec md5sum {} \; | sort',   # NOQA
+            use_sudo=True
+        ).stdout.splitlines() if not ('access' in f or 'patroni' in f)]
+        node_dump_filepaths = \
+            [logs_dump_filepaths['manager'][node.ip_address]] + \
+            [logs_dump_filepaths['db'][node.ip_address]] + \
+            [logs_dump_filepaths['broker'][node.ip_address]]
+        for dump_filepath in node_dump_filepaths:
+            local_dump_path = Path(tempfile.mkdtemp())
+            local_dump_filepath = local_dump_path / 'logs.tar'
+            cli_host.get_remote_file(dump_filepath, local_dump_filepath)
+            with tarfile.open(local_dump_filepath) as tar:
+                tar.extractall(local_dump_path)
+            files = list((local_dump_path / 'cloudify').rglob('*.*'))
+            mutable_files = ['cloudify/journalctl.log',
+                             'cloudify/nginx/cloudify.access.log',
+                             'cloudify/db_cluster/patroni/patroni.log']
+            for f in mutable_files:
+                assert local_dump_path / f in files
+            log_hashes_local = sorted(
+                [hashlib.md5(open(f, 'rb').read()).hexdigest() for f in
+                 files if not ('journalctl' in f.name or
+                               'access' in f.name or
+                               'patroni' in f.name or
+                               'startup_check' in f.name)])
+            assert set(log_hashes) == set(log_hashes_local)
+
+    logger.info('Testing `cfy logs backup`')
+    cli_host.run_command('{cfy} logs backup --verbose'.format(
+        cfy=paths['cfy']))
+    output = nodes[0].run_command('ls /var/log').stdout
+    assert 'cloudify-manager-logs_' in output
+
+    logger.info('Testing `cfy logs purge`')
+    cli_host.run_command('{cfy} logs purge --force'.format(cfy=paths['cfy']))
+
+    # Verify that each file under /var/log/cloudify is size zero
+    nodes[0].run_command(
+        'for f in $(sudo find /var/log/cloudify -type f -name "*") ; '
+        'do if $(test -s $f) ; then echo $f && exit 1 ; fi ; done'
```

Comment:
This isn't quite correct, you've inverted the condition.
However, this seems to work in my testing:
find /var/log/cloudify -type f -exec test -s {} \; -print -exec false {} +